taxi/views.py

```diff
@@ -1,47 +1,53 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
 from django.views import generic
 
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
```

Comment:
pass this logic to the util function and use it here

Reply:
I'm not quite sure what do you mean🤔